build/lib/pgBackRestBuild/Config/Data.pm

```diff
@@ -0,0 +1,2234 @@
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_RESTORE => {},
+        }
+    },
+
+    # LOG Section
+    #-------------------------------------------------------------------------------------------------------------------------------
+    &CFGBLDOPT_LOG_LEVEL_CONSOLE =>
+    {
+        &CFGBLDOPT_RULE_SECTION => CONFIG_SECTION_GLOBAL,
+        &CFGBLDOPT_RULE_TYPE => CFGBLDOPT_TYPE_STRING,
+        &CFGBLDOPT_RULE_DEFAULT => CFGBLDOPTDEF_DEFAULT_LOG_LEVEL_CONSOLE,
+        &CFGBLDOPT_RULE_ALLOW_LIST =>
+        {
+            lc(OFF)    => true,
+            lc(ERROR)  => true,
+            lc(WARN)   => true,
+            lc(INFO)   => true,
+            lc(DETAIL) => true,
+            lc(DEBUG)  => true,
+            lc(TRACE)  => true
+        },
+        &CFGBLDOPT_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_ARCHIVE_GET => {},
+            &CFGBLDCMD_ARCHIVE_PUSH => {},
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_CHECK => {},
+            &CFGBLDCMD_EXPIRE => {},
+            &CFGBLDCMD_INFO => {},
+            &CFGBLDCMD_RESTORE => {},
+            &CFGBLDCMD_STANZA_CREATE => {},
+            &CFGBLDCMD_STANZA_UPGRADE => {},
+            &CFGBLDCMD_START => {},
+            &CFGBLDCMD_STOP => {},
+        }
+    },
+
+    &CFGBLDOPT_LOG_LEVEL_FILE =>
+    {
+        &CFGBLDOPT_RULE_SECTION => CONFIG_SECTION_GLOBAL,
+        &CFGBLDOPT_RULE_TYPE => CFGBLDOPT_TYPE_STRING,
+        &CFGBLDOPT_RULE_DEFAULT => CFGBLDOPTDEF_DEFAULT_LOG_LEVEL_FILE,
+        &CFGBLDOPT_RULE_ALLOW_LIST =>
+        {
+            lc(OFF)    => true,
+            lc(ERROR)  => true,
+            lc(WARN)   => true,
+            lc(INFO)   => true,
+            lc(DEBUG)  => true,
+            lc(DETAIL) => true,
+            lc(TRACE)  => true
+        },
+        &CFGBLDOPT_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_ARCHIVE_GET => {},
+            &CFGBLDCMD_ARCHIVE_PUSH => {},
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_CHECK => {},
+            &CFGBLDCMD_EXPIRE => {},
+            &CFGBLDCMD_INFO => {},
+            &CFGBLDCMD_RESTORE => {},
+            &CFGBLDCMD_STANZA_CREATE => {},
+            &CFGBLDCMD_STANZA_UPGRADE => {},
+            &CFGBLDCMD_START => {},
+            &CFGBLDCMD_STOP => {},
+        }
+    },
+
+    &CFGBLDOPT_LOG_LEVEL_STDERR =>
+    {
+        &CFGBLDOPT_RULE_SECTION => CONFIG_SECTION_GLOBAL,
+        &CFGBLDOPT_RULE_TYPE => CFGBLDOPT_TYPE_STRING,
+        &CFGBLDOPT_RULE_DEFAULT => CFGBLDOPTDEF_DEFAULT_LOG_LEVEL_STDERR,
+        &CFGBLDOPT_RULE_ALLOW_LIST =>
+        {
+            lc(OFF)    => true,
+            lc(ERROR)  => true,
+            lc(WARN)   => true,
+            lc(INFO)   => true,
+            lc(DETAIL) => true,
+            lc(DEBUG)  => true,
+            lc(TRACE)  => true
+        },
+        &CFGBLDOPT_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_ARCHIVE_GET => {},
+            &CFGBLDCMD_ARCHIVE_PUS
```

Comment:
I can see why you added option_rule_required but I don't understand why &OPTION_RULE_DEFAULT => BACKREST_BIN,was removed.